src/org/opensha/commons/calc/magScalingRelations/magScalingRelImpl/TMG2017_MagAreaRel.java

```diff
@@ -0,0 +1,304 @@
+    	  return  0.121;
+    	else
+    	  // normal
+    	  return  0.181;
+      else
+    	// else subduction-interface
+      	return 0.150;
+    }
+   
+    public double getMagStdDev(Double rake,String regime) {
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return getMagStdDev(); 
+    }
+
+      
+    /** 
+     * overload to additionally consider regime 
+     */
+    public double getMedianMag(double area,  double rake, String regime){	 
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return this.getMedianMag(area);
+      }
+      
+    
+    
+    /**
+     * Computes the median rupture area from magnitude (for the previously set
+     * rake and regime values).
+     * @param mag - moment magnitude
+     * @return median area in km
+     */
+    public double getMedianArea(double mag){
+        if  (Double.isNaN(rake))
+           // this estimate is not available 
+           return Double.NaN;
+        if (regime.equalsIgnoreCase("crustal"))
+        	if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+        		// strike slip
+        		return  Math.pow(10.0, -3.486 + 0.942*mag);
+        	else if (rake > 0)
+        		// shallow thrust/reverse
+        		return  Math.pow(10.0, -4.362 + 1.049*mag);
+        	else
+        		// normal
+        		return  Math.pow(10.0, -2.551 + 0.808*mag);
+        else
+        	//subduction-interface
+        	return Math.pow(10.0, -3.292 + 0.949*mag);
+
+    }
+
+   
+    /**
+     * Computes the standard deviation of log(area) (base-10) from magnitude
+     *  (for the previously set rake and regime values)
+     * @return standard deviation
+     */
+    public double getAreaStdDev() {
+    	return getMagStdDev(); 
+
+    }
+
+    public double getAreaStdDev(Double rake,String regime) {
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return getMagStdDev(); 
+    }
+
+    
+    /** 
+     * overload to additionally consider regime 
+     */
+    public double getMedianArea(double mag,  double rake, String regime){	 
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return this.getMedianArea(mag);
+      }
+      
+    
+    /**
+     * Returns the name of the object
+     *
+     */
+    public String getName() {
+    	String type;
+        if (Double.isNaN(rake))
+            type = "not available";
+        
+        if (regime.equalsIgnoreCase("crustal"))
+          if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+            type = "strike-slip";
+          else if (rake > 0)
+            type = "shallow reverse-faulting";
+          else
+            type =  "normal-faulting";
+        else
+        	type = "interface";
+    	
+      return NAME +" for "+type+" events";
+    }
+    
+/*
+    // A quick test; 
+    // tested against sceqsrc (https://github.com/thingbaijam/sceqsrc)
+    public static void main(String args[]) {
+      TMG2017_MagAreaRel magRel = new TMG2017_MagAreaRel();
```

Comment:
@thingbaijam I see you've done some testing externally, but ideally the code is tested here with meaningful unit tests. I imagine you'd have to import some 'expected' data in a fixture and test that your class produces matching values.

Reply:
I agree, and will get a test data and implement unit testing.